torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -315,21 +316,34 @@ xla::XlaOp CreateIndex(const xla::XlaOp& input, const xla::XlaOp& indices) {
   xla::GatherDimensionNumbers dim_numbers;
   std::vector<xla::int64> slice_sizes;
   slice_sizes.reserve(input_shape.rank());
+  xla::int64 indices_rank = indices_shape.rank() - 1;
   for (xla::int64 i = 0; i < input_shape.rank(); ++i) {
     if (i < num_index_dims) {
       dim_numbers.add_collapsed_slice_dims(i);
       slice_sizes.push_back(1);
     } else {
       slice_sizes.push_back(input_shape.dimensions(i));
-      xla::int64 indices_rank = indices_shape.rank() - 1;
-      dim_numbers.add_offset_dims(i + indices_rank - num_index_dims);
+      dim_numbers.add_offset_dims(i - num_index_dims + indices_rank);
     }
   }
   dim_numbers.set_index_vector_dim(indices_shape.rank() - 1);
   for (xla::int64 i = 0; i < num_index_dims; i++) {
     dim_numbers.add_start_index_map(i);
   }
-  return xla::Gather(input, indices, dim_numbers, slice_sizes);
+  xla::XlaOp gathered = xla::Gather(input, indices, dim_numbers, slice_sizes);
```

Comment:
In these case I think it is more readable like:
```
result = foo(...);
if (cond) {
  ...
  result = moo(...);
}
return result;
```

Reply:
Done it with just a `Gather`, we can avoid the permute by making good use of `start_index_map`.